route/pool.go

```diff
@@ -371,11 +377,16 @@ func (p *EndpointPool) removeEndpoint(e *endpointElem) {
 	p.Update()
 }
 
-func (p *EndpointPool) Endpoints(logger logger.Logger, defaultLoadBalance string, initial string, mustBeSticky bool, azPreference string, az string) EndpointIterator {
-	switch defaultLoadBalance {
+func (p *EndpointPool) Endpoints(logger logger.Logger, initial string, mustBeSticky bool, azPreference string, az string) EndpointIterator {
+	switch p.LoadBalancingAlgorithm {
 	case config.LOAD_BALANCE_LC:
+		logger.Debug("endpoint-with-least-connection-lb-algo-added-to-pool")
```

Comment:
This function does not add something to the pool, it returns an iterator for the pool. Please adjust the log messages accordingly.

Reply:
done